x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/support/WatcherIndexTemplateRegistry.java

```diff
@@ -127,6 +141,52 @@ public void onFailure(Exception e) {
         });
     }
 
+    // Package visible for testing
+    LifecyclePolicy loadWatcherHistoryPolicy() {
+        return LifecyclePolicyUtils.loadPolicy(POLICY_WATCH_HISTORY.policyName, POLICY_WATCH_HISTORY.fileName, xContentRegistry);
+    }
+
+    private void addIndexLifecyclePolicyIfMissing(ClusterState state) {
```

Comment:
Since this only happens when the local node is master, it should only get called once with the atomic `compareAndSet` below. Also any subsequent calls would see that the policy already existed and not replace it

Reply:
ahh.. thanks, I missed this will only run on the master.

Reply:
minor nit: it is not only the master node, but is also put when this node is newer than the master node. The reason for this is, that with regards to the watcher history, a node that is not the master node might be updated first in a cluster, and that one could write the watch history in a format which requires a new watch history template.

I am not sure if this applies to ILM as well, as this is an administrative task compared to the fact that the watch history is written by non master nodes.